sdk/go/pulumi/context.go

```diff
@@ -675,7 +695,10 @@ func applyTransformations(t, name string, props Input, resource Resource, opts [
 
 		res := transformation(args)
 		if res != nil {
-			resOptions := merge(res.Opts...)
+			resOptions, err := tryMergeWithoutAwaiting(res.Opts...)
```

Comment:
Currently will fail if asked to await, however we can switch this to await here. This runs within contexts that are resolving resources already.